javatests/arcs/core/analysis/BoundedAbstractElementTest.kt

```diff
@@ -0,0 +1,126 @@
+package arcs.core.analysis
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+
+/** Tests for BoundedAbstractValue. */
+@RunWith(JUnit4::class)
+class BoundedAbstractElementTest {
+    private val top = BoundedAbstractElement.getTop<Int>()
+    private val bottom = BoundedAbstractElement.getBottom<Int>()
+    private val nine = BoundedAbstractElement.makeValue<Int>(9)
+    private val ten = BoundedAbstractElement.makeValue<Int>(10)
+
+    @Test
+    fun factoryMethodsAndPredicatesAreCorrect() {
+        with(top) {
```

Comment:
It would be slightly preferable to split this method up into three separate methods, one testing each case.

It's ever so slightly more code, but it's a bit more readable.

```
@Test
fun factoryMethodsAndPredicatesAreCorrect_top() {
    ...
}

@Test
fun factoryMethodsAndPredicatesAreCorrect_bottom() {
    ...
}

...
```

Reply:
Done.